src/interactive.js

```diff
@@ -1,75 +1,183 @@
 class Interact {
-  constructor(todos) {
-    this.todos = todos;
+  constructor(form, newData, input) {
+    this.form = form;
+    this.newData = newData;
+    this.input = input;
   }
 
-  displaylist() {
-    const todoList = JSON.parse(localStorage.getItem('todos'));
+  data() {
     const list = document.querySelector('#todo-data');
-    if (todoList === null) {
-      this.todos.forEach((task) => {
-        const li = document.createElement('li');
+    const todoList = JSON.parse(localStorage.getItem('todos'));
+    this.form.addEventListener('submit', (e) => {
+      e.preventDefault();
+      if (todoList === null) {
+        this.newData.push({
+          todoText: this.input.value,
+          completed: false,
+          todoIndex: this.newData.length,
+        });
+      } else {
+        this.newData = todoList;
+        this.newData.push({
+          todoText: this.input.value,
+          completed: false,
+          todoIndex: this.newData.length,
+        });
+      }
+
+      const todos = this.newData.sort((a, b) => {
+        const indexA = a.todoIndex;
+        const indexB = b.todoIndex;
+
+        if (indexA < indexB) {
+          return -1;
+        }
+        if (indexA > indexB) {
+          return 1;
+        }
+        return 0;
+      });
+      const li = document.createElement('li');
+
+      todos.forEach((task) => {
         const text = `<div class="list-container"> <input class='check-input' type='checkbox' value='${task.completed}'>
-                <p class="todo-text">${task.itemText}</p><a class="del-menu" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+                <p class="todo-text" contenteditable="true">${task.todoText}</p><div class="del-menu"><a class="edit" href="#">
+                <i class="fas fa-edit edit-box"></i></a><a href="#" class="del"><i class="fas fa-trash del-box"></i></a></div></div>`;
         li.classList.add('list-item');
         li.innerHTML = text;
-        list.appendChild(li);
       });
-    } else {
+
+      list.appendChild(li);
+      this.form.reset();
+      this.addItem(todos);
+      this.editItem();
+      const check = document.querySelectorAll('.check-input');
+      check.forEach((check, index) => {
+        check.addEventListener('change', (e) => {
+          e.preventDefault();
+          const divWrapper = check.parentElement;
+          if (check.checked) {
+            divWrapper.classList.add('checked');
+            check.checked = true;
+            todos[index].completed = true;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          } else {
+            divWrapper.classList.remove('checked');
+            todos[index].completed = false;
+            localStorage.setItem('todos', JSON.stringify(todos));
+          }
+        });
+      });
+      this.options();
```

Comment:
- [ ] Every time I add a new task, the browser highlights an error in this line of your code, kindly fix this. 🐱
   ![imagen](https://user-images.githubusercontent.com/60789317/140563375-2e43b898-ee50-49b2-b1a6-5b032de61e2a.png)